MVP/src/Output/CLI.ts

```diff
@@ -1,11 +1,109 @@
 
-const convertNDJSONObjToStr = (NDJSONObj: NDJSONRow) => {
-    return JSON.stringify(NDJSONObj);
-};
-
+/**
+ * Writes NDJSON formatted repository data to the CLI.
+ * @param repos - Array of repositories containing NDJSON rows.
+ */
 export const writeNDJSONToCLI = <T>(repos: Repository<T>[]) => {
     repos.forEach((repo) => {
         console.log(convertNDJSONObjToStr(repo.NDJSONRow));
     });
 };
+
+/**
+ * Prints the total count of dependencies and devDependencies from the package.json file.
+ */
+export const printDependencyCount = () => {
+    const packageJsonPath = path.resolve(__dirname, '../../package.json');
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
+    const dependencies = packageJson.dependencies || {};
+    const devDependencies = packageJson.devDependencies || {};
+    const totalDependencies = Object.keys(dependencies).length + Object.keys(devDependencies).length;
+    console.log(`\n${totalDependencies} dependencies installed...`);
+};
+
+/**
+ * Prints the test results from the output.txt file.
+ * If the file does not exist, it logs 'Test results not found.'
+ */
+export const printTestResults = () => {
+    const outputPath = path.resolve(__dirname, '../../coverage/output.txt');
+    const coverageSummaryPath = path.resolve(__dirname, '../../coverage/coverage-summary.json');
+
+    if (fs.existsSync(outputPath) && fs.existsSync(coverageSummaryPath)) {
+        const outputContent = fs.readFileSync(outputPath, 'utf-8');
+        const coverageSummary = JSON.parse(fs.readFileSync(coverageSummaryPath, 'utf-8'));
+
+        const testResultsLine = outputContent.split('\n').find((line) => line.includes('Tests:'));
+        if (testResultsLine) {
+            const match = testResultsLine.match(/Tests:\s+(\d+)\s+failed,\s+(\d+)\s+passed,\s+(\d+)\s+total/);
+            if (match) {
+                const failedTests = parseInt(match[1], 10);
+                const passedTests = parseInt(match[2], 10);
+                const totalTests = parseInt(match[3], 10);
+                const totalCoverage = ((passedTests / totalTests) * 100).toFixed(0);
+
+                console.log(`Total: ${totalTests}`);
+                console.log(`Passed: ${passedTests}`);
+                console.log(`Coverage: ${totalCoverage}%`);
+                console.log(
+                    `${passedTests}/${totalTests} test cases passed. ${totalCoverage}% line coverage achieved.`
+                );
+            } else {
+                console.log('Test results line not in expected format.');
+            }
+        } else {
+            console.log('Test results line not found.');
+        }
+    } else {
+        console.log('Test results not found.');
+    }
+};
+
+/**
+ * Processes an input file containing URLs, formats them into objects with initial scores, and prints each object as a JSON string.
+ * @param filePath - Path to the input file.
+ */
+export const processInputFile = (filePath: string) => {
```

Comment:
handled from Dorian's sanitize file

Reply:
at least partially (reading the file), if this is for outputting the array of repositories then that's another story

Then you will sort based on the information stored in repo.NDJSONRow.